core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1514,10 +1514,7 @@ class ReplicaManager(val config: KafkaConfig,
    * @return true if the request topic id is consistent, false otherwise
    */
   private def hasConsistentTopicId(requestTopicIdOpt: Option[Uuid], logTopicIdOpt: Option[Uuid]): Boolean = {
-    requestTopicIdOpt match {
-      case None => true
-      case Some(requestTopicId) => logTopicIdOpt.isEmpty || logTopicIdOpt.contains(requestTopicId)
-    }
+    requestTopicIdOpt.forall(requestTopicId => logTopicIdOpt.isEmpty || logTopicIdOpt.contains(requestTopicId))
```

Comment:
FYI reviewers -

No new object allocation here inside Option.forAll.

Implementation of `Option.forall`
```
@inline final def forall(p: A => Boolean): Boolean = isEmpty || p(this.get)
```